s2a.go

```diff
@@ -390,9 +391,21 @@ func NewS2ADialTLSContextFunc(opts *ClientOptions) func(ctx context.Context, net
 		}
 		timeoutCtx, cancel := context.WithTimeout(ctx, v2.GetS2ATimeout())
 		defer cancel()
-		s2aTLSConfig, err := factory.Build(timeoutCtx, &TLSClientConfigOptions{
-			ServerName: serverName,
-		})
+
+		var s2aTLSConfig *tls.Config
+		retryer := retry.NewRetryer()
+		for {
+			s2aTLSConfig, err = factory.Build(timeoutCtx, &TLSClientConfigOptions{
```

Comment:
Shouldn't we be checking the context to see if we've hit the timeout before re-entering? Similarly throughout.

Reply:
check added, thanks

Reply:
For my own understanding, why are we using 2 different contexts? Namely, `ctx` and `timeoutCtx`.

Reply:
should use the the same context, thanks for catching this.